File: java-code/src/main/java/refactoring/conditional_logic/replace_conditional_with_polymorphism/example1/after/EuropeanSwallow.java
package refactoring.conditional_logic.replace_conditional_with_polymorphism.example1.after;


public class EuropeanSwallow extends Bird {

  public EuropeanSwallow(Bird bird) {
    super(bird.getName(), bird.getType(), bird.getNumberOfCoconuts(), bird.getVoltage(), bird.isNailed());
  }

  @Override
  public String plumage() {
    return "average";
  }

  @Override
  public int airSpeedVelocity() {
    return 35;
  }

}

File: go-code/conditional_logic/replace_conditional_with_polymorphism/example1/after/EuropeanSwallow.go
package after

type EuropeanSwallow struct {
	BirdAttributes
}

func (e EuropeanSwallow) GetBirdAttributes() BirdAttributes {
	return e.BirdAttributes
}

func (e EuropeanSwallow) Plumage() string {
	return "average"
}

func (e EuropeanSwallow) AirSpeedVelocity() int {
	return 35
}
